backend/drm/drm.c

```diff
@@ -566,6 +566,31 @@ static bool drm_connector_set_mode(struct wlr_output *output,
 	return false;
 }
 
+
+bool wlr_drm_connector_add_mode(struct wlr_output *output, drmModeModeInfo *modeinfo) {
+	struct wlr_drm_connector *conn = (struct wlr_drm_connector *)output;
+	struct wlr_drm_mode *mode = calloc(1, sizeof(*mode));
+
+	if (!mode) {
+		return false;
+	}
+
+	assert(modeinfo);
+	memcpy (&mode->drm_mode, modeinfo, sizeof(*modeinfo));
+
+	mode->wlr_mode.width = mode->drm_mode.hdisplay;
+	mode->wlr_mode.height = mode->drm_mode.vdisplay;
+	mode->wlr_mode.refresh = mode->drm_mode.vrefresh;
+
+	wlr_log(L_INFO, "Registered custom mode "
+			"%"PRId32"x%"PRId32"@%"PRId32,
+			mode->wlr_mode.width, mode->wlr_mode.height,
+			mode->wlr_mode.refresh);
+	wl_list_insert(&conn->output.modes, &mode->wlr_mode.link);
+	return true;
+}
+
```

Comment:
There are extra blank lines on both sides of this function.

Reply:
bump